code/items/equipments/HealthkitEquipment.cs

```diff
@@ -0,0 +1,62 @@
+using Sandbox;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Items
+{
+    /// <summary>
+    /// Healthkit equipment definition, for the physical entity, see items/equipments/entities/HealthKitEntity.cs
+    /// </summary>
+    [Library("ttt_healthkit"), Hammer.Skip]
+    public partial class HealthkitEquipment : TTTEquipment, IBuyableItem
+    {
+        public override string ViewModelPath => "";
+        public override SlotType SlotType => SlotType.UtilityEquipment;
+
+        private const int DropPositionOffset = 50;
+        private const int DropVelocity = 500;
+
+        public int Price => 0;
+
+        public HealthkitEquipment() : base()
+        {
+
+        }
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            RenderAlpha = 0f;
+        }
+
+        public override void Simulate(Client client)
+        {
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (Owner is not TTTPlayer owner)
+            {
+                return;
+            }
+
+            using (Prediction.Off())
+            {
+                if (Input.Pressed(InputButton.Attack1))
+                {
+                    _ = new HealthkitEntity()
+                    {
+                        Position = Owner.EyePos + Owner.EyeRot.Forward * DropPositionOffset,
+                        Rotation = Owner.EyeRot,
+                        Velocity = Owner.EyeRot.Forward * DropVelocity,
+                    };
+                    (owner.Inventory as Inventory).Remove(this);
```

Comment:
I've seen this similar code three times, I wonder if we could write a utility method for it. It may end up being cleaner.

Reply:
How do you feel about the following?

Decoy.cs
```
            using (Prediction.Off())
            {
                if (Input.Pressed(InputButton.Attack1))
                {
                    (owner.Inventory as Inventory).DropEntity(this, typeof(DecoyEntity));
                }
            }
```

Inventory.cs
```
        public bool DropEntity(Entity self, Type entity)
        {
            Entity droppedEntity = Utils.GetObjectByType(entity);
            droppedEntity.Position = Owner.EyePos + Owner.EyeRot.Forward * DropPositionOffset;
            droppedEntity.Rotation = Owner.EyeRot;
            droppedEntity.Velocity = Owner.EyeRot.Forward * DropVelocity;
            return Remove(self);
        }
```
DropPositionOffset and DropVelocity are set at the top.

Reply:
Yeah I think this is fine, might be good to just do the initial inventory cast above for readability.

```csharp
Inventory inventory = owner.Inventory as Inventory
inventory.DropEntity(...)
```